sched/task/task_init.c

```diff
@@ -103,6 +104,14 @@ int nxtask_init(FAR struct task_tcb_s *tcb, const char *name, int priority,
       return ret;
     }
 
+  /* Duplicate the parent tasks environment */
+
+  ret = group_setupenv(tcb, envp);
```

Comment:
let change group_allocate to:
```
int  group_allocate(FAR struct task_tcb_s *tcb, uint8_t ttype, FAR char * const envp[]);
```
to catch all possible wrong usage places(e.g. nxtask_setup_vfork can be catch now).

Reply:
Or maybe add group_setupenv to nxtask_setup_vfork. I kind of like removing env_dup from group_allocate. Otherwise the name should be group_allocate_and_dup_env(), because it does more than simply allocates a group.

But this is just my thinking, I understand if there is a differing opinion.

Reply:
Both is fine to me.

Reply:
Done